### ISSUE TITLE
localhost:3001 refuses to connect. possible bug in Ruby interpreter or extension libraries.

### ISSUE THREAD

**Issue by Kishaihd** [tresacton/dspectrumgui#13]
### Issue Type
[bug]

### Actual Behaviour
What happened instead?
When running 
`rails s -p 3001`, 
it initially says 

```
rails s -p 3001
=> Booting Puma
=> Rails 4.2.5 application starting in development on http://localhost:3001
=> Run 'rails server -h' for more startup options
=> Ctrl-C to shutdown server
Puma 2.16.0 starting...
* Min threads: 0, max threads: 16
* Environment: development
* Listening on tcp://localhost:3001
```

However, when I open the browser (Google Chrome), it says "localhost refused to connect", and the terminal says 

```
Started GET "/" for ::1 at 2018-01-09 08:38:22 -0700
  ActiveRecord::SchemaMigration Load (0.2ms)  SELECT "schema_migrations".* FROM "schema_migrations"
Processing by DevicesController#index as HTML
Completed 401 Unauthorized in 19ms (ActiveRecord: 0.0ms)

Started GET "/users/sign_in" for ::1 at 2018-01-09 08:38:23 -0700
Processing by Devise::SessionsController#new as HTML
  Rendered devise/shared/_links.html.erb (10.2ms)
  Rendered devise/sessions/new.html.haml within layouts/application (68.0ms)
/home/kaden/radio/tools/dspectrumgui/app/views/layouts/application.html.haml:36: warning: duplicated key at line 36 ignored: :style
/home/kaden/radio/tools/dspectrumgui/app/views/layouts/application.html.haml:37: warning: duplicated key at line 37 ignored: :style
/home/kaden/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.5/lib/active_support/core_ext/marshal.rb:6: [BUG] Segmentation fault at 0x00000000000000
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-linux]
```
(Note the **[BUG] Segmentation fault at 0x00000000000000** above)
Then the ruby debug/crash report is printed, after which the terminal displays 

```
[NOTE]
You may have encountered a bug in the Ruby interpreter or extension libraries.
Bug reports are welcome.
For details: http://www.ruby-lang.org/bugreport.html

[1]    23103 abort (core dumped)  rails s -p 3001
```
### What commit were you running? 
(Go to your DSpectrumGUI app, and click "About", and paste the relevant information from the "Application Status Section" here)
App does not open. Cloned repository is latest from github as of 9 January 2018.

**Comment by tresacton** [comment]
I have been able to replicate this issue. I'm looking into it now

**Comment by tresacton** [comment]
Issue seems to be because of changes to the behaviour of a dependency in rails v2.2.2. Tested an upgrade to 2.2.8 which didn't have this problem. I have made an update. Please run the following and let me know if this has solved the problem for you as well.

```cd ~/dspectrum_gui # modify this if you've cloned DSpectrumGUI somewhere else
git fetch
git checkout master
git pull origin 

# launch the app again
rails s -p 3001
```

**Comment by Kishaihd** [comment]
I ran the commands you suggested, but then it told me `Your Ruby version is 2.2.2, but your Gemfile specified 2.2.8`
So I ran `rvm install 2.2.8` but that failed, saying 
`...
ruby-2.2.8 - #compiling - please wait
Error running '__rvm_make -j4',
There has been an error while running make. Halting the installation.`

**Comment by tresacton** [comment]
You've done the right thing with running rvm install 2.2.8. You need to install a more recent version of Ruby. I've set it to 2.2.8 in the Gemfile and .ruby-version file.

Your options:

1. Try another Ruby patch version, such as 2.2.7 and if it compiles on your system then you can update the Gemfile and .ruby-version to match that version number, then run the usual setup commands:
```
gem install bundler
bundle config build.nokogiri --use-system-libraries
bundle
rake db:setup && rake db:migrate && rake db:seed
rails s
```

2. Post an issue at https://github.com/rvm/rvm/issues/ as this is an RVM issue with your specific environment and they are better suited to assist you with determining the cause of the issue

3. Try to compile Ruby 2.2.8 using some other Ruby version manager or by itself (not recommended if you need to maintain/use multiple versions of Ruby on your machine).

4. Use a known-working operating system as a base, such as Kali or otherwise recommended in the Wiki

Out of interest, what OS are you using? I just tested this on Kali when I pushed the version update and didn't encounter the issue you are describing.

**Comment by Kishaihd** [comment]
I'm running Arch Linux. After a lot of messing around, it seems to be working!
I removed the Gemfile.lock, then reset my computer. When I logged back in, running `rvm install 2.2.8` worked, but running `bundle` kept giving me errors with puma. (puma had been giving me trouble since the beginning). So I changed the Gemfile.lock to allow for the latest version of puma, and `bundle` worked without any problems. Now when I run `rails s -p 3001` it serves the page just fine, and I'm able to log in and everything. :)
So basically:
Restarting my system made rvm work.
Changing the required puma to the latest version (3.11.0) make everything work.
